sound/soc/soc-dapm.c

```diff
@@ -1246,7 +1246,8 @@ int snd_soc_dapm_dai_get_connected_widgets(struct snd_soc_dai *dai, int stream,
 				custom_stop_condition);
 
 	/* Drop starting point */
-	list_del(widgets.next);
+	if (!list_is_singular(&widgets))
```

Comment:
What happens if you have more than one widget between widget and BE, it's valid and could be constructed.

Reply:
This part is a bit unclear to me. The problem I was facing was that the snd_soc_dapm_dai_get_connected_widgets() returned a list with only the BE DAI in the case of the tone pipeline and dropping the starting point resulted in an empty list of paths to process. 

the dai argument passed to this method was the cpu_dai->playback_widget() which is the BE dai in my case. And this was the only way I could get it to be enabled. I could use some help here to figure out the right way to do it.

Reply:
Yes, what happens if you have tone followed by a volume control?

Reply:
We do have a tone followed by volume in my topology. But the snd_soc_dapm_dai_get_connected_widgets() is called starting with the dai->playback_widget as the starting point. And this is the BE DAI.

Reply:
I think we should also state in a comment what we are doing here and why.